.github/workflows/dotnet.yaml

```diff
@@ -159,14 +159,20 @@ jobs:
     runs-on: ubuntu-latest
     if: ${{ !github.event.repository.fork && github.ref == 'refs/heads/main' }}
     needs: [all-required-checks-complete]
-    environment: main-deploy
-    permissions:
-      contents: write
+    environment: release
```

Comment:
Note the change of environment. The new environment has creds for a GitHub bot in.